### PR TITLE
MueLu: Create cuBLAS and cuSPARSE handles up front

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/muelu 

## Motivation
Create Kokkos-Kernels TPL handles up front, so that they don't mess up solve timings.

### CODE REVIEW

**Review by GrahamBenHarper — DISMISSED** [review]
lgtm

File: packages/muelu/test/unit_tests/MueLu_Test_ETI.hpp [hunk 1]
```diff
@@ -95,6 +100,18 @@ bool Automatic_Test_ETI(int argc, char *argv[]) {
...
+  // up timings.
+#ifdef HAVE_MUELU_KOKKOSKERNELS
+  KokkosKernels::Experimental::Controls controls;
+# ifdef KOKKOSKERNELS_ENABLE_TPL_CUBLAS
+  controls.getCublasHandle();
+# endif
+# ifdef KOKKOSKERNELS_ENABLE_TPL_CUSPARSE
+  controls.getCusparseHandle();
+# endif
+#endif
```
Comment (lucbv) [inline packages/muelu/test/unit_tests/MueLu_Test_ETI.hpp:114]: You could consider adding a fence here for good measure, I believe that cuSPARSE is doing some async memory touches in the handle.

**Review by lucbv — APPROVED** [review]
This is fine

File: packages/muelu/test/unit_tests/MueLu_Test_ETI.hpp (hunk 1 above)
Comment (cgcgcg) [reply packages/muelu/test/unit_tests/MueLu_Test_ETI.hpp:114]: Ok, will do. In my runs it seemed to not matter, but maybe I'm hitting an early fence somewhere else..

**Review by jhux2 — APPROVED** [review]
lgtm, thanks!

**Pull request merged**